src/kvstore.c

```diff
@@ -583,13 +643,43 @@ void kvstoreIteratorRelease(kvstoreIterator *kvs_it) {
     hashtableIterator *iter = &kvs_it->di;
     hashtableResetIterator(iter);
     /* In the safe iterator context, we may delete entries. */
-    freeHashtableIfNeeded(kvs_it->kvs, kvs_it->didx);
+    if (kvs_it->didx != -1) {
```

Comment:
nit. -1 is a magic number. let's either check for validity or use the new symbolic name (`KVSTORE_INDEX_NOT_FOUND`) you introduced in this pr :)
```suggestion
    if (kvs_it->didx >= 0) {
```

Reply:
This is an existing artifact of the kvstoreIter implementation (using -1). I updated here but I also generally feel like refactoring these things should be pulled into dedicated PRs